src/pubsub/mod_pubsub_db_mnesia.erl

```diff
@@ -200,11 +227,41 @@ del_node(Nidx) ->
 -spec set_node(mod_pubsub:pubsubNode()) -> {ok, mod_pubsub:nodeIdx()}.
 set_node(#pubsub_node{id = undefined} = Node) ->
     CreateNode = Node#pubsub_node{id = pubsub_index:new(node)},
-    set_node(CreateNode);
-set_node(#pubsub_node{id = Nidx} = Node) ->
+    OldParents = [],
+    set_node(CreateNode, OldParents);
+set_node(Node) ->
+    OldParents = get_parentnodes_names(Node),
+    set_node(Node, OldParents).
+
+set_node(#pubsub_node{id = Nidx} = Node, OldParents) ->
     mnesia:write(Node),
+    set_subnodes(Node, OldParents),
     {ok, Nidx}.
 
+get_parentnodes_names(#pubsub_node{nodeid = NodeId}) ->
+    case mnesia:read(pubsub_node, NodeId, write) of
+        [] ->
+            [];
+        [#pubsub_node{parents = Parents}] ->
+            Parents
+    end.
+
+set_subnodes(#pubsub_node{parents = Parents}, Parents) ->
+    ok;
+set_subnodes(#pubsub_node{nodeid = NodeId, parents = NewParents}, OldParents) ->
+    set_subnodes(NodeId, NewParents, OldParents).
+
+set_subnodes({Key, Node}, NewParents, OldParents) ->
+    Deleted = OldParents -- NewParents,
```

Comment:
Are duplicates possible here, how about using `sets` instead of `lists`?

Reply:
Good idea

Reply:
done.

Reply:
thanks 👍